### PR TITLE
feat(nvim): add nvim dotfiles with lua support

### PR DESCRIPTION
**Pull request by airscripts**
I've made this PR that converts your .vimrc into an init.lua for your neovim.  
Please, make sure it works properly as I have only made a few tests focusing specifically on the porting.  

I've even structured your neovim's dotfiles folder as a wish for your journey with this cool piece of software.  
Make sure to add some awesome plugins!

### CODE REVIEW

File: .config/nvim/core/keybindings.lua [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+map("n", "N", "J")
+map("n", "O", "P")
+map("n", "P", "R")
+map("n", "R", "S")
+map("n", "S", "D")
+map("n", "T", "F")
+map("n", "U", "I")
+map("n", "Y", "O")
+map("n", "Y" '"+y')
+map("v", "Y" '"+y')
```
Comment (mateonunez) [inline .config/nvim/core/keybindings.lua]: A coma is missing after `"Y"`.

File: .config/nvim/core/keybindings.lua [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+map("n", "L", "U")
+map("n", "N", "J")
+map("n", "O", "P")
+map("n", "P", "R")
+map("n", "R", "S")
+map("n", "S", "D")
+map("n", "T", "F")
+map("n", "U", "I")
+map("n", "Y", "O")
+map("n", "Y" '"+y')
```
Comment (mateonunez) [inline .config/nvim/core/keybindings.lua]: A coma is missing after `"Y"`

File: .config/nvim/core/keybindings.lua [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+map("v", "Y" '"+y')
+map("n", "D", '"+d')
+map("v", "D", '"+d')
+map("n", "dd", "gg")
+map("n", "df", "ge")
+map("n", "dF", "gE")
+map("n", "dD" '^"+d$')
```
Comment (mateonunez) [inline .config/nvim/core/keybindings.lua]: A coma is missing after `"dD"`.

File: .config/nvim/core/init.lua [hunk 4]
```diff
@@ -0,0 +1 @@
+require("core.keybindings")
```
Comment (mateonunez) [inline .config/nvim/core/init.lua]: I think we need to import `core.options` too here, WDYT?

**Review by mateonunez — COMMENTED** [review]
Hey @Airscripts! Thanks for this amazing Pull Request. I've tested the configuration in my local environment.  I left you a couple of comments, let me know what do you think about it.

Also, it seems that we need to change the folder structure.
The current folder structure to load configurations files is:
> .config/nvim/core

Reading the documentation it seems we need to change it to:

> .config/nvim/**lua**/core

Otherwise `nvim` wouldn't load the config Lua files.

I'd really appreciate this and I want to merge it as soon as possible. Let me know if I can help you with anything.

File: .config/nvim/core/keybindings.lua (hunk 1 above)
Comment (airscripts) [reply .config/nvim/core/keybindings.lua]: Added!

File: .config/nvim/core/keybindings.lua (hunk 2 above)
Comment (airscripts) [reply .config/nvim/core/keybindings.lua]: Added!

File: .config/nvim/core/keybindings.lua (hunk 3 above)
Comment (airscripts) [reply .config/nvim/core/keybindings.lua]: Added!

File: .config/nvim/core/init.lua (hunk 4 above)
Comment (airscripts) [reply .config/nvim/core/init.lua]: You're right. I've added it in f4ca17b.

**Comment by airscripts** [comment]
It may be useful to create a simple Dockerfile in order to test the configurations in a clean environment.  
I've had this idea in my head for a while for my dotfiles too.

**Review by mateonunez — APPROVED** [review]

**Pull request merged**